### ISSUE TITLE
Can't Read New Diagrams.net XML

### ISSUE THREAD

**Issue by herryliu** [dmulyalin/N2G#11]
I have tried to read the XML file saved from Diagrams.net. It shows zero nodes and edges but it were objects in the file.

Seems the Diagrams.net uses format like following to describe the node.

        <mxCell id="GfU-AREGhvKuv_HTweEu-1" value="R1" style="rounded=0;whiteSpace=wrap;html=1;" vertex="1" parent="1">
          <mxGeometry x="320" y="340" width="120" height="60" as="geometry" />
        </mxCell>
If I read the source code, it expect in following format:
` 
<object id="{id}" label="{label}">
      <mxCell style="{style}" vertex="1" parent="1">
          <mxGeometry x="{x_pos}" y="{y_pos}" width="{width}" height="{height}" as="geometry"/>
      </mxCell>
    </object>
  `
There is no <object> tag from Diagrams.net's export. Not sure if that is issue.

**Comment by AndreasWeier** [comment]
I can repoduce your issue. Possible solution:
~~~
    def from_xml(self, text_data):
        """
        Method to load graph from .drawio XML text produced by DrawIO

        **Args**

            * text_data - text data to load
        """
        self.drawing = ET.fromstring(text_data)
        # extract diagrams, nodes, edges IDs
        for diagram_elem in self.drawing.findall("./diagram"):
            diagram_id=diagram_elem.attrib["id"]
            self.nodes_ids.setdefault(diagram_id, [])
            self.edges_ids.setdefault(diagram_id, [])

            # iterate over mxcells to extract nodes and edges
            for object in diagram_elem.findall("./mxGraphModel/root/UserObject"):
                self._extract_node_and_edge(object, diagram_id)

            # iterate over mxcells to extract nodes and edges
            for object in diagram_elem.findall("./mxGraphModel/root/object"):
                self._extract_node_and_edge(object, diagram_id)

        self.go_to_diagram(diagram_index=0)

    def _extract_node_and_edge(self, xmlobj, diagram_id):
        object_id = xmlobj.attrib["id"]
        mxCell = xmlobj.find("./mxCell")
        # check if this is the edge
        if "source" in mxCell.attrib and "target" in mxCell.attrib:
            self.edges_ids[diagram_id].append(object_id)
        else:
            self.nodes_ids[diagram_id].append(object_id)
~~~

**Comment by dmulyalin** [comment]
Thank you for raising this issue, this should be addressed in 3594e00f382667288ae5907eb616227508d60ee5
